datalad/local/gitcredential_datalad.py

```diff
@@ -0,0 +1,181 @@
+import sys
+
+from datalad.downloaders import UserPassword
+from datalad.downloaders.providers import (
+    AUTHENTICATION_TYPES,
+    Providers,
+)
+from datalad.local.gitcredential import _credspec2dict
+from datalad.utils import Path
+
+git_credential_datalad_help = """\
+Git credential interface to DataLad's credential management system.
+
+Only DataLad's UserPassword-type credentials are supported. This helper
+passes standard 'get', 'store', and 'erase' actions on to the respective
+interfaces in DataLad.
+
+usage: git credential-datalad [option] <action>
+
+options:
+    -h                   Show this help.
+    --non-interactive    Don't ask for user confirmation when storing to
+                         DataLad's credential system. This may fail if default
+                         names result in a conflict with existing ones.
+                         This mode is used for DataLad's CI tests.
```

Comment:
I would prefer to not expose such switches in the API. An envvar should be as, or even more, flexible in terms of triggering test-related special modes inside some other application.

Reply:
@bpoldrack Are you planning on changing this? Or is there a need for it?